nuxeo-services/nuxeo-platform-usermanager-core/src/main/java/org/nuxeo/ecm/platform/usermanager/io/NuxeoGroupJsonReader.java

```diff
@@ -88,29 +90,43 @@ public NuxeoGroupJsonReader() {
     protected NuxeoGroup readEntity(JsonNode jn) throws IOException {
         GroupConfig groupConfig = userManager.getGroupConfig();
         String id = getStringField(jn, "id");
-        DocumentModel groupModel = null;
-        if (id == null) {
-            // backward compatibility
-            id = getStringField(jn, "groupname");
+        String groupName = getStringField(jn, "groupname");
+        if (StringUtils.isBlank(id) || ((StringUtils.isNotBlank(groupName)) && !id.equals(groupName))) {
+            // backward compatibility if `id` not found or if `groupname` is different
+            id = groupName;
         }
-        if (id != null) {
+
+        DocumentModel groupModel = null;
+        if (StringUtils.isNotBlank(id)) {
             groupModel = userManager.getGroupModel(id);
+
         }
         if (groupModel == null) {
             groupModel = userManager.getBareGroupModel();
             groupModel.setProperty(groupConfig.schemaName, groupConfig.idField, id);
         }
 
+        String beforeReadLabel = (String) groupModel.getProperty(groupConfig.schemaName, groupConfig.labelField);
+
         readProperties(groupModel, groupConfig, jn);
         readMemberUsers(groupModel, groupConfig, jn);
         readMemberGroups(groupModel, groupConfig, jn);
         readParentGroups(groupModel, groupConfig, jn);
 
-        String label = (String) groupModel.getProperty(groupConfig.schemaName, groupConfig.labelField);
-        if (label == null) {
-            // backward compatibility
-            label = getStringField(jn, "grouplabel");
-            groupModel.setProperty(groupConfig.schemaName, groupConfig.labelField, label);
+        // override the `groupname` that may have been in the `properties` object
+        if (StringUtils.isNotBlank(id)) {
+            groupModel.setProperty(groupConfig.schemaName, groupConfig.idField, id);
+        }
+
+        // override with the compatibility `grouplabel` if needed
+        if (jn.has("grouplabel")) {
+            String compatLabel = getStringField(jn, "grouplabel");
```

Comment:
It would be nicer to use constants for `"grouplabel"` and `"groupname"`, and factor them between reader and writer.